docs/gallery/plot_external_resources.py

```diff
@@ -196,20 +202,21 @@
 ###############################################################################
 # Using the add_ref method with a field
 # ------------------------------------------------------
-# In the above example, the ``field`` keyword argument was empty because the data
-# of the :py:class:`~hdmf.container.Data` object passed in for the ``container``
-# argument was being associated with a resource. However, you may want to associate
-# an attribute of a :py:class:`~hdmf.container.Data` object with a resource, or
-# a dataset or attribute of a :py:class:`~hdmf.container.Container` object with
-# a resource. To disambiguate between these different fields, you can set the
-# 'field' keyword.
+# It is important to keep in mind that the :py:class:`~hdmf.common.resources.Object` object
+# we add to the :py:class:`~hdmf.common.resources.ObjectTable` is the object
+# holding the attribute or the closest parent object. Recall that the objects in the
```

Comment:
```suggestion
# It is important to keep in mind that the :py:class:`~hdmf.common.resources.Object.object_id` object
# we add to the :py:class:`~hdmf.common.resources.ObjectTable` is the object
# holding the attribute or the closest parent object. Recall that the objects in the
```

Doesn't the `hdmf.common.resources.Object` refer to the full object (i.e, including object_id, relative_path, and field), not just the closest parent? The closest parent is `hdmf.common.resources.Object.object_id`

Reply:
hdmf.common.resources.Object does refer to the full Object. I was saying how the Object needs to be the closest/holding the attribute. hdmf.common.resources.Object.object_id is the id of that object.

Reply:
In that case we should maybe say:

It is important to keep in mind that when adding and  :py:class:`~hdmf.common.resources.Object`` to the  :py:class:`~hdmf.common.resources.ObjectTable`, the parent object identified by :py:class:`~hdmf.common.resources.Object.object_id` must be the closest parent to the target object (i.e., :py:class:`~hdmf.common.resources.Object.relative_path` must be the shortest possible path and as such cannot contain any objects with a `data_type` and associated `object_id`)